src/NexusMods.App.UI/Pages/ModInfo/ModInfoViewModel.cs

```diff
@@ -0,0 +1,96 @@
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.Loadouts.Mods;
+using NexusMods.App.UI.Controls.ModInfo.Error;
+using NexusMods.App.UI.Controls.ModInfo.Loading;
+using NexusMods.App.UI.Controls.ModInfo.ViewModFiles;
+using NexusMods.App.UI.Pages.ModInfo.Types;
+using NexusMods.App.UI.Resources;
+using NexusMods.App.UI.Windows;
+using NexusMods.App.UI.WorkspaceSystem;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+
+namespace NexusMods.App.UI.Pages.ModInfo;
+
+public class ModInfoViewModel : APageViewModel<IModInfoViewModel>, IModInfoViewModel
+{
+    [Reactive]
+    public LoadoutId LoadoutId { get; set; }
+    
+    [Reactive]
+    public ModId ModId { get; set; }
+    
+    [Reactive]
+    public CurrentModInfoPage Page { get; set; }
+
+    [Reactive]
+    public IViewModelInterface PageViewModel { get; set; } = default!;
+
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILoadoutRegistry _registry;
+    private bool _isInvalid;
+
+    public ModInfoViewModel(IWindowManager windowManager, IServiceProvider serviceProvider, ILoadoutRegistry registry) : base(windowManager)
+    {
+        _serviceProvider = serviceProvider;
+        _registry = registry;
+        this.WhenActivated(delegate(CompositeDisposable dp)
+        {
+            GetWorkspaceController().SetTabTitle(GetModName(out _isInvalid), WorkspaceId, PanelId, TabId);
+            PageViewModel = !_isInvalid ? new DummyLoadingViewModel() : new DummyErrorViewModel();
+            if (!_isInvalid)
+            {
+                this.WhenAnyValue(x => x.Page)
+                    .OffUi()
+                    .Select(CreateNewPage)
+                    .OnUI()
+                    .Subscribe(x => PageViewModel = x)
+                    .DisposeWith(dp);
+            }
```

Comment:
This currently would recreate the viewmodel every time the ModInfo selected content is changed. 
Ideally we would only create the viewModel the first time.